docs/configure

```diff
@@ -647,33 +647,32 @@ virtual_forward:
 
 virtual_domains:
   driver = redirect
+  address_data = ${lookup mysql{\
+        select smtp, users.sa_tag*10 AS sa_tag, users.on_spamassassin AS on_spamassassin \
+        from users,domains \
+        where localpart = '${quote_mysql:$local_part}' \
+                and domain = '${quote_mysql:$domain}' \
+                and domains.enabled = '1' \
+                and users.enabled = '1' \
+                and users.domain_id = domains.domain_id}{$value}fail}
   allow_fail
-  data = ${lookup mysql{select smtp from users,domains \
-  		where localpart = '${quote_mysql:$local_part}' \
-		and domain = '${quote_mysql:$domain}' \
-		and domains.enabled = '1' \
-		and users.enabled = '1' \
-		and users.domain_id = domains.domain_id}}
-  headers_add = ${if >{$spam_score_int}{${lookup mysql{select users.sa_tag * 10 from users,domains \
-  		where localpart = '${quote_mysql:$local_part}' \
-		and domain = '${quote_mysql:$domain}' \
-		and users.on_spamassassin = '1' \
-		and users.domain_id=domains.domain_id }{$value}fail}} {X-Spam-Flag: YES\n}{} }
-  # You may have to use single colons in headers_remove rule, or comment it out altogether
-  # if using Exim v. 4.85 or older. See https://github.com/avleen/vexim2/issues/21 and
-  # check out the links pasted in that issue for more info.
+  data = ${extract{smtp}{$address_data}}
+  headers_add = ${if and { \
+                    {>{$spam_score_int}{${extract{sa_tag}{$address_data}}}} \
+                    {eq{1}{${extract{on_spamassassin}{$address_data}}}} \
+                    } {X-Spam-Flag: YES\n}{} }
+  headers_remove = ${if or { { <{$spam_score_int}{1} } \
+                            { and {{<{$spam_score_int}{${extract{sa_tag}{$address_data}}} } \
+                                   {eq {1}{${extract{on_spamassassin}{$address_data}}}} \
+                                  } \
+                            } { eq {0}{${extract{on_spamassassin}{$address_data}}}} \
```

Comment:
I have only rewritten the rules we had before. But yes, we remove these 
headers, if the user turns off spamassassin. Headers from the sender's 
mail server are probably deleted as well. I have seen that some 
organizations change their spam-headers to `X-Spam-Organization:` that 
they won't be deleted by following mail servers.

Am 2015-10-31 14:54, schrieb Rimas Kudelis:
> In docs/configure [1]:
> 
>> -        and domain = '${quote_mysql:$domain}' \
>> -        and users.on_spamassassin = '1' \
>> -        and users.domain_id=domains.domain_id }{$value}fail}}
> {X-Spam-Flag: YES\n}{} }
>> -  # You may have to use single colons in headers_remove rule, or
> comment it out altogether
>> -  # if using Exim v. 4.85 or older. See
> https://github.com/avleen/vexim2/issues/21 and
>> -  # check out the links pasted in that issue for more info.
>> +  data = ${extract{smtp}{$address_data}}
>> +  headers_add = ${if and { \
>> +
> {>{$spam_score_int}{${extract{sa_tag}{$address_data}}}} \
>> +
> {eq{1}{${extract{on_spamassassin}{$address_data}}}} \
>> +                    } {X-Spam-Flag: YES\n}{} }
>> +  headers_remove = ${if or { { <{$spam_score_int}{1} } \
>> +                            { and
> {{<{$spam_score_int}{${extract{sa_tag}{$address_data}}} } \
>> +                                   {eq
> {1}{${extract{on_spamassassin}{$address_data}}}} \
>> +                                  } \
>> +                            } { eq
> {0}{${extract{on_spamassassin}{$address_data}}}} \
> 
> Do I understand it correctly that you want to remove X-Spam-Score if
> on_spamassassin is 0? Why? Was it definitely us who added that header?
> 
> 
> --
> Reply to this email directly or view it on GitHub [2].
> 
> 	*
> 
> Links:
> ------
> [1] https://github.com/vexim/vexim2/pull/82#discussion_r43571912
> [2] https://github.com/vexim/vexim2/pull/82/files#r43571912